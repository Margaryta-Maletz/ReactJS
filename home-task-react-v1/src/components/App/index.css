.wrapper {
    width: 1200px;
    box-sizing: border-box;
    padding: 20px 60px;
    display: flex;
}
