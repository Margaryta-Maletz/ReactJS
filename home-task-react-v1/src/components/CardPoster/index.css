.card-poster_wrapper {
    width: 322px;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

    font-family: Montserrat;
    font-style: normal;
    font-weight: 500;
    color: white;
    mix-blend-mode: normal;
}

.card-poster_title {
    font-size: 18px;
    line-height: 22px;
    opacity: 0.7;
    margin: 22px 0 8px;
    width: 75%;
}

.card-poster_genre {
    font-size: 14px;
    line-height: 17px;
    opacity: 0.5;
    margin-block: 0;
    margin: 0 0 50px;
}

.card-poster_release-date {
    width: 66px;
    height: 26px;
    border: 1px solid #555555;
    box-sizing: border-box;
    border-radius: 4px;
    font-size: 14px;
    line-height: 24px;
    text-align: center;
    margin: 22px 0 8px;
}
