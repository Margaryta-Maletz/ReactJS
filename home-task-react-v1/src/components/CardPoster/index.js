import React from 'react';

export const CardPoster = (props) => {
    return (
        <div className='card-poster_wrapper'>
            <img className='card-poster_image' src={ props.cards.imagePath } alt={ props.cards.imageAlt } />
            <h4 className='card-poster_title'>{ props.cards.title }</h4>
            <p className='card-poster_genre'>{ props.cards.genre }</p>
            <p className='card-poster_release-date'>{ props.cards.releaseDate }</p>
        </div>
    )
}
