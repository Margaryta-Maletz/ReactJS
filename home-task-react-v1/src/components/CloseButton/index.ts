export { CloseButton, CloseButtonSize } from './CloseButton';
