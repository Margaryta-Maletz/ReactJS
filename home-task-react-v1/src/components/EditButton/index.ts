export { EditButton, CardPosterProp } from './EditButton';
