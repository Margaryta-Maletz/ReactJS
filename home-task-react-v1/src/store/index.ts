import thunk from 'redux-thunk';
import {configureStore, getDefaultMiddleware, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './slice';

const middleware = getDefaultMiddleware({
    imitableCheck: false,
    serializableCheck: false,
    thunk: true,
});

export const store = configureStore({
    reducer: rootReducer,
    middleware: middleware,
    devTools: process.env.NODE_ENV !== 'production',
});

/*export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;*/
