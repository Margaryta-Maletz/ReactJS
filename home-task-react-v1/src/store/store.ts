import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './reducers';

export const store = createStore(
    rootReducer,
    applyMiddleware(thunk)
);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;
