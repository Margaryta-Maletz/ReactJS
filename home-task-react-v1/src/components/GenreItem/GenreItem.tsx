import React from "react";
import styles from './GenreItem.module.css';
/*import {useDispatch, useSelector} from "react-redux";
import {IState} from "../../store/types";
import {setActiveGenre} from "../../store/slice";*/
import {Genre, genres} from "../../consts";
// import {useHistory, useLocation} from "react-router-dom";

interface GenreItemProp {
    title?: string;
    value: string | null;
}

export const GenreItem: React.FC<GenreItemProp> = ({ title, value }) => {
/*    const dispatch = useDispatch();
    const location = useLocation();
    const history = useHistory();
    const { activeGenre } = useSelector((state: IState) => state);*/

    const handleSelectGenre = () => {
/*        dispatch(setActiveGenre(value));
        const params = new URLSearchParams(location.search);
        value ? params.set('genre', value) : params.delete('genre');
        params && history.push(`?${params.toString()}`);*/
    }

    return (
        <>
            {value === null
                ? <li className={`${styles.item} ${styles.active}`}>{title || value}</li>
                : <li className={styles.item} onClick={handleSelectGenre}>{title || value}</li>}
        </>
    )
}
