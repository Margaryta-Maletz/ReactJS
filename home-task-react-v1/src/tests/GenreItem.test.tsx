import React from "react";
import { render } from '@testing-library/react';
import { GenreItem } from '../components/GenreItem/';
import {Provider} from "react-redux";
import {store} from "../store/store";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
        pathname: "localhost:8080/search/"
    }),
    useHistory: (path: string) => ({
        push: jest.fn(),
        location: {
            pathname: "localhost:8080/search/" + path
        }
    })
}));

describe('make snapshot', () => {
    it('renders', () => {
        const { container } = render(
            <Provider store={store}>
                <GenreItem title="All" value={null}/>
            </Provider>);
        expect(container).toMatchSnapshot();
    });
});
