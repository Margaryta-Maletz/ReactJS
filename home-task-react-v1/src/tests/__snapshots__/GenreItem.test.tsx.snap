// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make snapshot renders 1`] = `
<div>
  <li
    class="genre_item active"
  >
    All
  </li>
</div>
`;
